enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+  global-admission-hsts EP, which defines a cluster-scoped API, using which the cluster admin can specify HSTS parameters that may or must be specified on routes
+  for a given domain: https://github.com/openshift/api/blob/6ba31fa438f20d6c822ef47ad3d771309cb5216b/config/v1/types_ingress.go#L64-L87.
+  
+  HSTS should be configurable by a cluster admin for a domain, and not override-able by a route annotation or by a custom header setting.
+
+#### Why do we need the kubebuilder validation tags on the Route CRD in openshift/api in addition to validation in library-go?
+  Answer: The library-go validation is used for server-side validation whereas the OpenAPI schema on the Route CRD may be used for client-side validation.
+  On OCP, kube-apiserver delegates the Route API to openshift-apiserver, which employs the library-go validation to validate Route objects (see <https://github.com/openshift/openshift-apiserver/blob/aac3dd5bf0547e928103a0f718ca104b1bb13930/pkg/route/apis/route/validation/validation.go>).
+  MicroShift does not use openshift-apiserver; instead, on MicroShift, kube-apiserver handles the Route API using the Route CRD. 
+    
+  However, kube-apiserver has an admission plugin for the Route CRD that employs the same library-go validation that openshift-apiserver uses (see <https://github.com/openshift/kubernetes/blob/15f19ea2dd700767e5337502aec753d2a6e26905/openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go>).
+  Defining validation for the same API in both library-go and the OpenAPI schema introduces the risk that someone could inadvertently introduce discrepancies between the library-go validation and the OpenAPI schema, leading to buggy and potentially confusing behavior.  Care must be taken to avoid introducing such discrepancies.
+  Historical note: At one time, MicroShift relied on the OpenAPI schema for server-side validation of the Route CRD, and for this reason, kubebuilder validation tags were added to the
+  Route CRD (see <https://github.com/openshift/api/pull/1228>).
+
+  Since then, an admission plugin has been added to kube-apiserver that validates the Route CRD using the same
+  library-go validation on MicroShift that openshift-apiserver uses on OpenShift (see <https://github.com/openshift/kubernetes/pull/1416>).
+  As a result, the kubebuilder validation tags on the Route CRD are no longer required specifically for MicroShift; however, the kubebuilder validation tags are still required for client-side validation.
+
+#### Why don't you set/delete the headers on frontend public_ssl in the haproxy.config?
+  Answer: The connection is still encrypted in public_ssl, so it is not possible to examine or modify the HTTP session there. The fe_sni and fe_no_sni frontends terminate TLS.
+
+#### Can we add conditions for haproxy to check to set the value? For example: Can I set the value `"%[res.hdr(server)] if { req.hdr(user-agent) -m sub evil}"` for `x-frame-options` ?
```

Comment:
This shows up wrong in the md view, it is missing the word "add".  Something needs to be fixed in the formatting?

```suggestion
#### Can we use conditional values for headers?
```